core/provider/service.go

```diff
@@ -602,40 +608,51 @@ func (s *Service) getResources(ctx context.Context, p *domain.Provider) ([]*doma
 			filteredResources = append(filteredResources, r)
 		}
 	}
-
 	flattenedProviderResources := flattenResources(filteredResources)
+
 	existingProviderResources := map[string]bool{}
-	for _, r := range flattenedProviderResources {
-		for _, er := range existingGuardianResources {
-			if er.Type == r.Type && er.URN == r.URN {
-				if existingDetails := er.Details; existingDetails != nil {
-					if r.Details != nil {
+	updatedResources := []*domain.Resource{}
+	for _, newResource := range flattenedProviderResources {
+		found := false
+		for _, existingResource := range existingGuardianResources {
+			if existingResource.Type == newResource.Type && existingResource.URN == newResource.URN {
+				found = true
+				if existingDetails := existingResource.Details; existingDetails != nil {
+					if newResource.Details != nil {
 						for key, value := range existingDetails {
-							if _, ok := r.Details[key]; !ok {
-								r.Details[key] = value
+							if _, ok := newResource.Details[key]; !ok {
+								newResource.Details[key] = value
 							}
 						}
 					} else {
-						r.Details = existingDetails
+						newResource.Details = existingDetails
+					}
+					if isUpdated, diff := compareResources(*existingResource, *newResource); isUpdated {
+						s.logger.Debug(ctx, "diff", "resources", diff)
+						updatedResources = append(updatedResources, newResource)
+						s.logger.Info(ctx, "resources is updated", "resource", newResource.Name)
 					}
 				}
-				existingProviderResources[er.ID] = true
+				existingProviderResources[existingResource.ID] = true
 				break
 			}
 		}
+		if !found {
+			updatedResources = append(updatedResources, newResource)
```

Comment:
is it duplicated with https://github.com/goto/guardian/pull/184/files#diff-90d0af1332cbe67a6e33fa3414e20536758c05060bb29eba832e08cf813a9fb6R632?

Reply:
No, this is to check if there is new resource added.